File: apps/api/app/schemas/bug.py
from typing import Literal

from pydantic import BaseModel

from app.schemas.pagination import PaginatedResponse

BugSeverity = Literal["Critical", "High", "Medium", "Low"]
BugStatus = Literal["New", "Triaging", "In progress", "Fixed", "Closed"]


class BugReportCreate(BaseModel):
    title: str
    customer: str
    product_area: str
    severity: BugSeverity
    status: BugStatus
    assignee: str
    source: str
    reproduction_steps: str
    linked_release: str
    reported_at: str


class BugReportUpdate(BaseModel):
    title: str | None = None
    customer: str | None = None
    product_area: str | None = None
    severity: BugSeverity | None = None
    status: BugStatus | None = None
    assignee: str | None = None
    source: str | None = None
    reproduction_steps: str | None = None
    linked_release: str | None = None
    reported_at: str | None = None


class BugReportItem(BaseModel):
    id: str
    title: str
    customer: str
    product_area: str
    severity: BugSeverity
    status: BugStatus
    assignee: str
    source: str
    reproduction_steps: str
    linked_release: str
    reported_at: str

    model_config = {"from_attributes": True}


BugReportListResponse = PaginatedResponse[BugReportItem]

File: apps/web/src/types/bug.ts
import type { PaginatedResponse, PaginationParams } from "./pagination";
import type { SummaryItem } from "./summary";

type BugSeverity = "Critical" | "High" | "Medium" | "Low";
type BugStatus = "New" | "Triaging" | "In progress" | "Fixed" | "Closed";

type BugReport = {
  id: string;
  title: string;
  customer: string;
  productArea: string;
  severity: BugSeverity;
  status: BugStatus;
  assignee: string;
  source: string;
  reproductionSteps: string;
  linkedRelease: string;
  reportedAt: string;
};

type BugReportCreateInput = Omit<BugReport, "id">;

type BugReportUpdateInput = Partial<BugReportCreateInput>;

type BugFilters = {
  productArea?: string;
  search: string;
  severity?: BugSeverity;
  status?: BugStatus;
};

type BugReportListResponse = PaginatedResponse<BugReport>;
type BugSummary = SummaryItem;

export type {
  BugFilters,
  BugReport,
  BugReportCreateInput,
  BugReportListResponse,
  BugReportUpdateInput,
  BugSeverity,
  BugStatus,
  BugSummary,
  PaginationParams,
};
